engines/detect_modules.mk

```diff
@@ -0,0 +1,15 @@
+MODULE := detection
+
+DETECT_OBJS_DYNAMIC=$(addprefix ../,$(DETECT_OBJS))
+
+MODULE_OBJS := \
+	../engines/detection.o \
```

Comment:
This will not compile. Apparently, you had an empty directory `../detection`, this is why this path was still valid for you. More work is needed here for removing the `detection` prefix.